macros/does_table_exist.sql

```diff
@@ -1,17 +1,14 @@
 {%- macro does_table_exist(table_name) -%}
-    {%- if execute -%} -- returns true when dbt is in execute mode
-    {%- set ns = namespace(has_table=false) -%} -- declare boolean namespace and default value 
-        {%- for node in graph.sources.values() -%} -- grab sources from the dictionary of nodes 
-        -- call the database for the matching table
-            {%- set source_relation = adapter.get_relation(
-                    database=node.database,
-                    schema=node.schema,
-                    identifier=node.identifier ) -%} 
-            {%- if source_relation == None and node.name | lower == table_name | lower -%} 
-                {{ return(False) }} -- return false if relation identified by the database.schema.identifier does not exist for the given table name
```

Comment:
What is the/is there a benefit of searching the graph? 

To me it makes more sense to directly the values for `adapter.get_relation` since we have them, [which is used in Lever](https://github.com/fivetran/dbt_lever_source/blob/main/macros/does_table_exist.sql).